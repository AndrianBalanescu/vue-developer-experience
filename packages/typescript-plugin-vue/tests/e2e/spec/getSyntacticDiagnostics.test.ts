import Proto from 'typescript/lib/protocol'
import { createLanguageServerForTest } from '../server'

describe('getSyntacticDiagnostics', () => {
  test('should report syntax issues in a vue file', async () => {
    const server = createLanguageServerForTest()
    server.openFile('fixture-syntactic-diagnostics.vue')
    server.sendCommand(Proto.CommandTypes.SyntacticDiagnosticsSync, <
      Proto.SyntacticDiagnosticsSyncRequestArgs
    >{
      file: server.canonicalFileName('fixture-syntactic-diagnostics.vue'),
    })
    await server.close()
    expect(server.responses).toHaveLength(1)
    const response: Proto.SyntacticDiagnosticsSyncResponse = server.responses[0]
    expect(response.command).toBe(Proto.CommandTypes.SyntacticDiagnosticsSync)
    expect(response.body).toHaveLength(2)
    expect(
      (response.body as Proto.Diagnostic[]).map((item) => [
        item.text,
        item.start,
      ])
    ).toMatchSnapshot()
  })
})
