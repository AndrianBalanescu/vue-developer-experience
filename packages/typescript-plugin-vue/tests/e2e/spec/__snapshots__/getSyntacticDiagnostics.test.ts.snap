// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getSyntacticDiagnostics should report syntax issues in a vue file 1`] = `
Array [
  Array [
    "Expression expected.",
    Object {
      "line": 4,
      "offset": 18,
    },
  ],
  Array [
    "',' expected.",
    Object {
      "line": 8,
      "offset": 3,
    },
  ],
]
`;
